src/components/Ideas/CompIdeaDetail.js

```diff
@@ -0,0 +1,45 @@
+import React, { useContext, useEffect, useState } from "react"
+import { IdeaContext } from "./IdeaProvider"
+import { useParams, useHistory } from "react-router-dom"
+import "./Idea.css"
+
+export const CompIdeaDetail = (props) => {
+    const { ideas, getCompleteIdeas, deleteIdea } = useContext (IdeaContext)
+    const [ idea, setIdea ] = useState({ artist: {} })
+
+    const { ideaId } = useParams()
+
+    useEffect(() => {
+        if(props.idea){
+            setIdea(props.idea)
+        } else {
+            const thisIdea = idea.find(i => i.id === parseInt(ideaId)) || { artist: {} }
+            setIdea(thisIdea)
+        }
+    }, [idea, ideas])
```

Comment:
SInce `idea` is being updated in this `useEffect`, this could be running more than it needs to, or even running in an infinite loop.

Reply:
I'd just look at `ideas` to mitigate that.